BLSQ/iaso
# Add Predefined filters to forms

In the mobile application (and later in the web interface), we'll have to be able to filter some people based on some JsonLogic, which can include the current time. It was discussed and decided with @madewulf that we should attach a predefined filter to the `Form` model so that this can be reused in Org Units and Instances alike.

It was decided that for the time being, it's OK not to make an API that allows you to add/modify those filters and just go through the admin.

Related JIRA tickets : WC2-155

## Self proofreading checklist


## Changes

A new `FormPredefinedFilter` entity has been created and linked to `Form`.
The form endpoints can now return those filters if requested through the `fields` query parameter.

## How to test

Apply the migration and make sure you have a form created.
Go in the admin and add a predefined filter to the form.
Call `api/mobile/forms/?fields=id,name,predefined_filters`, and the predefined filter you created should be returned with the `Form`.
